File: Examples/Cross Machine Platform/CS_WPF/OkumaInterface/OkumaSim.cs
namespace CS_WPF.OkumaInterface
{
    using System.Collections.Generic;

    public class OkumaSim : IOkuma
    {
        Dictionary<int, double> _variables;

        public OkumaSim()
        {
            _variables = new Dictionary<int, double>();

            for (int i = 1; i < 199; i++)
            {
                _variables.Add(i, 0D);
            }
        }

        public OperationModeEnum GetOperationMode()
        {
            return OperationModeEnum.Auto;
        }

        public double ReadVariable(int variableNumber)
        {
            return _variables[variableNumber];
        }

        public void WriteVariable(int variableNumber, double newValue)
        {
            _variables[variableNumber] = newValue;
        }

        public void Dispose()
        {
            _variables.Clear();
            _variables = null;
        }

    }
}

File: Examples/Cross Machine Platform/VB_WPF/OkumaInterface/OkumaSim.vb
Imports VB_WPF

Public Class OkumaSim
    Implements IOkuma

    Private _variables As Dictionary(Of Integer, Double)

    Public Sub New()
        _variables = New Dictionary(Of Integer, Double)()

        For index = 1 To 200
            _variables.Add(index, 0#)
        Next

    End Sub

    Public Sub WriteVariable(variableNum As Integer, newValue As Double) Implements IOkuma.WriteVariable
        _variables(variableNum) = newValue
    End Sub

    Public Function GetOperationMode() As OperationModeEnum Implements IOkuma.GetOperationMode
        Return OperationModeEnum.Auto
    End Function

    Public Function ReadVariable(variableNum As Integer) As Double Implements IOkuma.ReadVariable
        Return _variables(variableNum)
    End Function

    Public Sub Dispose() Implements IDisposable.Dispose
        _variables.Clear()
        _variables = Nothing
    End Sub
End Class
